lib/snippets/kubernetes-deploy

```diff
@@ -66,99 +200,455 @@ class KubernetesDeploy
       errors << "#{@template_path} doesn't contain valid templates (postfix .yml or .yml.erb)"
     end
 
-    if @namespace.nil? || @namespace.empty?
+    if @namespace.blank?
       errors << "Namespace must be specified"
     end
 
-    if @context.nil? || @context.empty?
+    if @context.blank?
       errors << "Context must be specified"
     end
 
     raise FatalDeploymentError, "Configuration invalid: #{errors.join(", ")}" unless errors.empty?
-    logger.info("All required parameters and files are present")
-  end
-
-  def apply_all_templates
-    found = false
-    Dir.foreach(@template_path) do |file|
-      file_path = "#{@template_path}/#{file}"
-      case File.extname(file)
-      when '.yml'
-        found ||= true
-        apply_template(file_path)
-      when '.erb'
-        found ||= true
-        render_and_apply_template(file_path)
-      end
-    end
-    raise FatalDeploymentError, "No templates found in #{@template_path}" unless found
+    KubernetesDeploy.logger.info("All required parameters and files are present")
   end
 
-  def render_and_apply_template(file_path)
-    erb_template = ERB.new(File.read(file_path))
-    erb_binding = binding
-    template_variables.each do |var_name, value|
-      erb_binding.local_variable_set(var_name, value)
+  def deploy_resources(resources, prune: false)
+    command = ["apply", "--namespace=#{@namespace}"]
+    KubernetesDeploy.logger.info("Deploying resources:")
+
+    resources.each do |r|
+      KubernetesDeploy.logger.info("- #{r.id}")
+      command.push("-f", r.file.path)
```

Comment:
So then instead of saving the rendered templates to `{tempfile}` we'd save them to `{type}/{tempfile}`? Seems like slightly more work--is there a concrete advantage to doing that?